util/weekly_build/05_BuildCache.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+set -ex
+
+if [ -z $SETUPDONE ]; then . ShellSetup.sh $* ; fi
+
+cd $RUNDIR/$RUNID
+
+set +x
+. setup.sh
+set -x
+
+for compiler in $COMPILERS; do
+  cd $RUNDIR/$RUNID/envs/build-${compiler/@/-}
+  spack env activate .
+  spack buildcache push --unsigned --force ${BUILD_CACHE} $PACKAGES_TO_INSTALL
+  spack buildcache rebuild-index ${BUILD_CACHE}
```

Comment:
Encountering an error here on Orion-intel

+ command spack buildcache push --unsigned --force /work/noaa/epic/role-epic/jenkins/workspace/spack-stack/build_auto/Orion-intel/build_cache
+ return 0
+ spack buildcache rebuild-index /work/noaa/epic/role-epic/jenkins/workspace/spack-stack/build_auto/Orion-intel/build_cache
+ : this is a shell function from: /work/noaa/epic/role-epic/jenkins/workspace/spack-stack/build_auto/Orion-intel/weekly/spack/share/spack/setup-env.sh
+ : the real spack script is here: /work/noaa/epic/role-epic/jenkins/workspace/spack-stack/build_auto/Orion-intel/weekly/spack/bin/spack
+ _spack_shell_wrapper buildcache rebuild-index /work/noaa/epic/role-epic/jenkins/workspace/spack-stack/build_auto/Orion-intel/build_cache
+ for var in DYLD_LIBRARY_PATH DYLD_FALLBACK_LIBRARY_PATH
+ eval 'if [ -n "${DYLD_LIBRARY_PATH-}" ]; then export SPACK_DYLD_LIBRARY_PATH=${DYLD_LIBRARY_PATH}; fi'
++ '[' -n '' ']'
+ for var in DYLD_LIBRARY_PATH DYLD_FALLBACK_LIBRARY_PATH
+ eval 'if [ -n "${DYLD_FALLBACK_LIBRARY_PATH-}" ]; then export SPACK_DYLD_FALLBACK_LIBRARY_PATH=${DYLD_FALLBACK_LIBRARY_PATH}; fi'
++ '[' -n '' ']'
+ '[' -n '' ']'
+ _sp_flags=
+ '[' '!' -z x ']'
+ '[' buildcache '!=' buildcache ']'
+ '[' -n '' ']'
+ '[' '' '!=' '' ']'
+ _sp_subcommand=
+ '[' '!' -z x ']'
+ _sp_subcommand=buildcache
+ shift
+ case $_sp_subcommand in
+ command spack buildcache rebuild-index /work/noaa/epic/role-epic/jenkins/workspace/spack-stack/build_auto/Orion-intel/build_cache
==> Warning: Encountered problem listing packages at file:///work/noaa/epic/role-epic/jenkins/workspace/spack-stack/build_auto/Orion-intel/build_cache/build_cache: [Errno 2] No such file or directory: '/work/noaa/epic/role-epic/jenkins/workspace/spack-stack/build_auto/Orion-intel/build_cache/build_cache'
==> Error: Unable to generate package index: Failed to get list of specs from file:///work/noaa/epic/role-epic/jenkins/workspace/spack-stack/build_auto/Orion-intel/build_cache/build_cache
++ ERROR
++ alert_cmd ./05_BuildCache.sh
++ echo 'Your run failed in ./05_BuildCache.sh. This is a placeholder alerting function. '\''alert_cmd'\'' should be defined for each system.'
Your run failed in ./05_BuildCache.sh. This is a placeholder alerting function. 'alert_cmd' should be defined for each system.
++ exit 1

Reply:
It appears that the build cache is empty because nothing got installed... That's odd that it didn't fail on the installation step. Looking at envs/build-intel/log.install, it seems it tried to install gmake but then doesn't show any further output. Are you setting $PACKAGES_TO_INSTALL?

Reply:
Trying again with:
```
export BUILD_CACHE=${workspace}
mkdir -p ${BUILD_CACHE}/build_cache
```